### PR TITLE
dev/test/prod parity copy and images

### PR DESCRIPTION
**Pull request by nzoschke**
screen shot 2016-10-27 at 12 47 54 pm
## Release Playbook
- [ ] Rebase against master
- [ ] Code review
- [ ] Merge into master
- [ ] Verify changes at http://site-staging.convox.com
- [ ] Promote release on `site-production` in the `convox/production` Rack

### CODE REVIEW

**Comment by nzoschke** [comment]
[Blog Content: Dev/Test/Prod Parity](https://trello.com/c/JiYDoR7R/913-blog-content-dev-test-prod-parity)

File: _posts/2016-10-28-dev-test-prod-parity.md [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+author: Noah Zoschke
+twitter: nzoschke
+---
+
+The [Twelve Factor App](https://12factor.net/) methodology is a series of best practices to follow when building your app to make it easy to deploy to the cloud.
+
+One of the factors is [Dev/Prod Parity](https://12factor.net/dev-prod-parity):
+
+> Keep development, staging, and production as similar as possible
+
```
Comment (mattmanning) [inline _posts/2016-10-28-dev-test-prod-parity.md]: This line break doesn't show up on the rendered page for some reason.

File: _posts/2016-10-28-dev-test-prod-parity.md [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+<!--more-->
+
+# What Parity?
+
+A very common setup is [Homebrew](http://brew.sh/) for development on a Mac laptop, [CircleCI](https://circleci.com/) for testing and [Heroku](https://www.heroku.com/) for production. With these systems, it’s hard to know, let alone guarantee, what version of Node.js, ImageMagick and Postgres are used in each environment.
+
+![Laptop, CI Service, and Production Service Incompatibility Matrix](/assets/img/dev-test-prod-matrix.png){: .center } *Laptop, CI Service, and Production Service Incompatibility Matrix*
+
+It’s not impossible to set it all up but there are pitfalls.
```
Comment (mattmanning) [inline _posts/2016-10-28-dev-test-prod-parity.md]: Needs a comma after "up".

File: _posts/2016-10-28-dev-test-prod-parity.md [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+
+You need to maintain good documentation and scripts in the app codebase to help maintain a Homebrew environment.
```
Comment (mattmanning) [inline _posts/2016-10-28-dev-test-prod-parity.md]: These 4 "You" points seem like they could be a bulleted list.

File: _posts/2016-10-28-dev-test-prod-parity.md [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+
+Once it works, it’s smooth sailing. Until you want to upgrade Ruby, or add PhantomJS for web acceptance testing, or Homebrew packages change, or...
+
+# Parity Begins in Development With Containers
+
+Containerization technology, made accessible by Docker, offers a better way.
+
+First, install the fantastic [Docker for Mac](https://docs.docker.com/engine/installation/mac/) to get a Docker environment running on your laptop. This app is practically magic, and offers a fast Linux environment on your Mac laptop.
+
+Next, put in some effort to "Dockerize" your app. The goal is to boot your app with a single command: `docker-compose up`. This involves writing a `Dockerfile` and a `docker-compose.yml` file.
```
Comment (mattmanning) [inline _posts/2016-10-28-dev-test-prod-parity.md]: Worth mentioning `convox start` as an alternative here?

File: _posts/2016-10-28-dev-test-prod-parity.md [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+        build: .
+        command: ["node", "worker.js"]
+        links:
+         - redis
+      redis:
+        image: redis
+
+Next, you need write a `Dockerfile` that can build and run your app. You need to pick a base operating system, add system dependencies, then add a few more commands to install the language packages your app uses.
+
+I recommend using Ubuntu 16.04. Ubuntu 16.04 makes it easy to get the latest Node.JS, Ruby, Python PHP system packages. It is a "Long Term Support" distribution that will receive security updates through 2021. This is another huge advantage of Docker: we are no longer constrained by the limited CircleCI and Heroku operating system choices.
```
Comment (mattmanning) [inline _posts/2016-10-28-dev-test-prod-parity.md]: I think this is missing an "or" or an "and" between "Python" and "PHP".

File: _posts/2016-10-28-dev-test-prod-parity.md (hunk 1 above)
Comment (nzoschke) [inline _posts/2016-10-28-dev-test-prod-parity.md]: Fixed in css.

File: _posts/2016-10-28-dev-test-prod-parity.md (hunk 4 above)
Comment (nzoschke) [inline _posts/2016-10-28-dev-test-prod-parity.md]: ok added it.

**Pull request merged**